analyzers/rspec/cs/S2222_c#.html

```diff
@@ -0,0 +1,70 @@
+<p>If a lock is acquired and released within a method, then it must be released along all execution paths of that method.</p>
+<p>Failing to do so will expose the conditional locking logic to the method’s callers and hence be deadlock-prone.</p>
+<p>The types tracked by the rule are: Monitor, Mutex, ReaderWriterLock, ReaderWriterLockSlim and SpinLock from the <code>System.Threading</code>
+namespace.</p>
+<h2>Noncompliant Code Example</h2>
+<pre>
+class MyClass
+{
+  private object obj = new object();
+
+  public void DoSomethingWithMonitor()
+  {
+    Monitor.Enter(obj); // Noncompliant
+    if (IsInitialized())
+    {
+      // ...
+      Monitor.Exit(obj);
+    }
+  }
+
+  private ReaderWriterLockSlim lockObj = new ReaderWriterLockSlim();
+
+  public void DoSomethingWithReaderWriteLockSlim()
+  {
+    lockObj.EnterReadLock(); // Noncompliant
```

Comment:
I think that if a lock doesn't get released in the method where it gets acquired, we should raise an issue...

Maybe update the description...

Reply:
I agree with Andrei, if the method just acquires the lock but does not release it, it might create deadlocks and unexpected behaviour for the consumers of that method.

Reply:
As discussed offline, S2222 should follow RSPEC. Not-releasing locks will be another rule (to specify)